argcomplete/scripts/activate_global_python_argcomplete.py

```diff
@@ -120,35 +120,40 @@ def link_user_rcfiles():
     append_to_config_file(bash_completion_user_file, bash_shellcode.format(activator=get_activator_path()))
 
 
-parser = argparse.ArgumentParser(description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter)
-parser.add_argument("-y", "--yes", help="automatically answer yes for all questions", action="store_true")
-parser.add_argument("--dest", help='Specify the shell completion modules directory to install into, or "-" for stdout')
-parser.add_argument("--user", help="Install into user directory", action="store_true")
-argcomplete.autocomplete(parser)
-args = parser.parse_args()
-destinations = []
-
-if args.dest:
-    if args.dest != "-" and not os.path.exists(args.dest):
-        parser.error(f"directory {args.dest} was specified via --dest, but it does not exist")
-    destinations.append(args.dest)
-elif site.ENABLE_USER_SITE and site.USER_SITE in argcomplete.__file__:
-    print(
-        "Argcomplete was installed in the user site local directory. Defaulting to user installation.", file=sys.stderr
-    )
-    link_user_rcfiles()
-elif sys.prefix != sys.base_prefix:
-    print("Argcomplete was installed in a virtual environment. Defaulting to user installation.", file=sys.stderr)
-    link_user_rcfiles()
-elif args.user:
-    link_user_rcfiles()
-else:
-    print("Defaulting to system-wide installation.", file=sys.stderr)
-    destinations.append(f"{get_zsh_system_dir()}/_python-argcomplete")
-    destinations.append(f"{get_bash_system_dir()}/python-argcomplete")
-
-for destination in destinations:
-    install_to_destination(destination)
-
-if args.dest is None:
-    print("Please restart your shell or source the installed file to activate it.", file=sys.stderr)
+def main():
```

Comment:
This breaks the script by moving variables that other functions depend on from global scope into functional scope.

Reply:
Fixed it, though there might be cleaner ways to do so.